### PR TITLE
Dice roll command

### PR DESCRIPTION
**Pull request by sbrugel**

### CODE REVIEW

File: src/commands/fun/diceroll.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Message } from 'discord.js';
+import { Command } from '@lib/types/Command';
+
+export default class extends Command {
+
+	description = 'Get a random number within a user-specified range (min and max inclusive). If no range is specified by the user, defaults to a range from 1 to 6. Also works for negative numbers!';
+	usage = '<min #> | <max #>';
```
Comment (BenSegal855) [inline src/commands/fun/diceroll.ts:7]: The convention is to use square brackets for optional args and pointy ones for required ones. Judging by `argParser()`, these should be square brackets.

**Review by BenSegal855 — CHANGES_REQUESTED** [review]

File: src/commands/fun/diceroll.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	aliases = ['random', 'rand'];
+
+	run(msg: Message, [minimum, maximum]: Array<number>): Promise<Message> {
+		return msg.channel.send(`Your random number is ${Math.floor((Math.random() * (maximum - minimum + 1)) + minimum)}`);
+	}
+
+	argParser(_msg: Message, input: string): Array<number> {
+		const [minimum, maximum] = input.split('|').map(arg => arg.trim());
+		if (!input) {
+			return [1, 6];
```
Comment (joshualyon7) [inline src/commands/fun/diceroll.ts:17]: Would be nice to have this as a constant up top
`const DEFAULT_RANGE = [1, 6];`

File: src/commands/fun/diceroll.ts (hunk 1 above)
Comment (sbrugel) [reply src/commands/fun/diceroll.ts]: Fixed now, thank you!

File: src/commands/fun/diceroll.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+	description = 'Get a random number within a user-specified range (min and max inclusive). If no range is specified by the user, defaults to a range from 1 to 6. Also works for negative numbers!';
+	usage = '[min #] | [max #]';
+	aliases = ['random', 'rand'];
+
+	run(msg: Message, [minimum, maximum]: Array<number>): Promise<Message> {
+		return msg.channel.send(`Your random number is ${Math.floor((Math.random() * (maximum - minimum + 1)) + minimum)}`);
+	}
+
+	argParser(_msg: Message, input: string): Array<number> {
```
Comment (joshualyon7) [inline src/commands/fun/diceroll.ts:14]: The arg parser should also check that what the user entered is indeed two numbers

**Review by joshualyon7 — CHANGES_REQUESTED** [review]

File: src/commands/fun/diceroll.ts (hunk 3 above)
Comment (sbrugel) [reply src/commands/fun/diceroll.ts:14]: This should be fixed via the isNaN function and a check for further args using `,...rest`

File: src/commands/fun/diceroll.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { Message } from 'discord.js';
+import { Command } from '@lib/types/Command';
+
```
[the remark this answers is not in the document]
Comment (sbrugel) [reply src/commands/fun/diceroll.ts:3]: Updated description to reflect this. However, the last sentence had to be removed as the full string including var names exceeded 200 characters.

**Review by joshualyon7 — DISMISSED** [review]

File: src/commands/fun/diceroll.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+const DEFAULT_RANGE = [1, 6];
+export default class extends Command {
+
+	description = `Get a random integer within a user-specified range (min and max inclusive). If no range is specified, defaults to a range from ${DEFAULT_RANGE[0]} to ${DEFAULT_RANGE[1]}.`;
+	usage = '[min #] | [max #]';
+	aliases = ['random', 'rand'];
```
Comment (renross) [inline src/commands/fun/diceroll.ts:9]: This is mildly unimportant but you might want to consider also having aliases somewhat similar to the actual command name (but shortened), like s;dice or s;roll.

**Review by renross — APPROVED** [review]

**Comment by bladetyr** [comment]
Looks fine to me other than the alias tidbit, which isn't a big deal anyway

**Review by bladetyr — APPROVED** [review]

**Review by joshualyon7 — APPROVED** [review]

**Pull request merged**